manifest/kcp.yaml

```diff
@@ -0,0 +1,151 @@
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kcp
+spec:
+  secretName: kcp-secret
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+      - redhat
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+    - kcp
+    - "*openshiftapps.com"
+  ipAddresses:
+    - 0.0.0.0
+  issuerRef:
+    name: kcp
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kcp
+spec:
+  ports:
+    - protocol: TCP
+      name: kcp
+      port: 6443
+      targetPort: 6443
+    - protocol: TCP
+      name: virtual-workspaces
+      port: 6444
+      targetPort: 6444
+  selector:
+    app: kcp
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kcp
+  labels:
+    app: kcp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kcp
+  template:
+    metadata:
+      labels:
+        app: kcp
+    spec:
+      containers:
+      - name: kcp
+        image: ghcr.io/kcp-dev/kcp:latest
+        ports:
+        - containerPort: 6443
+        command:
+        - /kcp
+        args:
+        - start
+        - --run-controllers
+        - --auto-publish-apis
+        - --push-mode
+        - --etcd-servers=https://etcd:2379
+        - --etcd-keyfile=/etc/etcd/tls/server/tls.key
+        - --etcd-certfile=/etc/etcd/tls/server/tls.crt
+        - --etcd-cafile=/etc/etcd/tls/server/ca.crt
+        - --tls-private-key-file=/etc/kcp/tls/server/tls.key
+        - --tls-cert-file=/etc/kcp/tls/server/tls.crt
+        - --root-directory=/tmp/kcp
+        - --oidc-issuer-url=https://sso.redhat.com/auth/realms/redhat-external
```

Comment:
could we point the public instance to github SSO ?

Reply:
Sure, I can work on setting that up.  Why the preference for github SSO?

Reply:
Any public one is fine. Not everybody in the community has RH SSO access :)

Reply:
We could actually do even better than that. Point to RHSSO and turn Github Social Login Provider in that specific Realm in Keycloak. This way, you could either login using RHSSO or just click "login with Github".

Reply:
That would be great, but that assumes we can modify the realm, right?

Reply:
Yes, that's right. I'm not sure if this is possible in our scenario but probably worth checking.

Reply:
I doubt that will be allowed now in sso.rh (may be in future). Enabling Github IDP in sso.rh means all rh site supports the github login which isn't the case now.

Reply:
I believe we've landed on RH SSO for now and we can follow up later with a separate instance that uses Github IDP.